### DISCUSSION TITLE
Embedding Maps

### DISCUSSION THREAD

**Discussion by jjzajdel** [open-sdg/open-sdg#1556] [Q&A]
Hi,

I am working on developing an Open SDG dashboard for the San Diego region. I have worked through the tutorial provided by Open SDG and I currently have some data hosted in the chart format.

The issue I am having concerns that mapping functionality. A priority for this project is to be able to host disaggregated geospatial data. Since the built in Open SDG mapping functionality doesn't allow for disaggregation, we used folium to construct a map that suits our needs and we have html code for this map. We would like to host on our Open SDG platform. 

Previously, we had had been hosting these data by creating a separate repository on GitHub, hosting an external webpage through GitHub Pages and then embedding that external page with an iframe and the embedded feature functionality of the Open SDG template. However, as this methodology would require a separate repository for each indicator, we would like to bring the html files/code into the data repository. 

So far I have tried using an iframe to call to an html file within the data repository using the soc functionality of iframes. However, when I do this I get a 404 error where the map should be. I have also tried to add the html code directly to the indicator yml file but that prevents the site from being built. 

Our data repository can be found here: https://github.com/jjzajdel/data
Our site repository can be found here: https://github.com/SANDAG/SDG
Our site can be found here: https://sandag.github.io/SDG/

Could you provide any guidance on how we might host html code with an iframe through the embedded feature functionality of the Open SDG template? Any assistance that you could provide would be greatly appreciated.

**Comment by brockfanning** [comment]
@jjzajdel The scripts that "build" the data repository generate specific files and publish them in predictable folder structures, but it does not publish arbitrary files. I think there are a couple ways to accomplish what you're looking for.

1. Put the files in the site repository instead of the data repository. The site repository *does* publish arbitrary files, so this should work if it's an acceptable solution for you.
2. Update your data repository's GitHub Actions workflow files (for example [this one](https://github.com/open-sdg/open-sdg-data-starter/blob/develop/.github/workflows/deploy-to-staging.yml)) to copy in your arbitrary files after the data repo has been built. I haven't tested this but I imagine it would be something like this:

```
    - name: Copy map files
      run: |
        cp -r maps _site
```

(that would go in between the "Build data" step and the "Deploy to GitHub Pages" step)
